src/NServiceBus.AwsLambda.SQS/TransportWrapper/ServerlessTransport.cs

```diff
@@ -1,26 +1,53 @@
 ﻿namespace NServiceBus.AwsLambda.SQS.TransportWrapper
 {
-    using Settings;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus;
     using Transport;
 
     class ServerlessTransport<TBaseTransport> : TransportDefinition
-        where TBaseTransport : TransportDefinition, new()
+        where TBaseTransport : TransportDefinition
     {
-        public ServerlessTransport()
+        // HINT: This constant is defined in NServiceBus but is not exposed
+        const string MainReceiverId = "Main";
+
+        public ServerlessTransport(TBaseTransport baseTransport)
+            : base(baseTransport.TransportTransactionMode, baseTransport.SupportsDelayedDelivery, baseTransport.SupportsPublishSubscribe, baseTransport.SupportsTTBR)
         {
-            baseTransport = new TBaseTransport();
+            BaseTransport = baseTransport;
         }
 
-        public override string ExampleConnectionStringForErrorMessage { get; } = string.Empty;
+        public TBaseTransport BaseTransport { get; }
 
-        public override bool RequiresConnectionString => baseTransport.RequiresConnectionString;
+        public PipelineInvoker PipelineInvoker { get; private set; }
 
-        public override TransportInfrastructure Initialize(SettingsHolder settings, string connectionString)
+        public override async Task<TransportInfrastructure> Initialize(HostSettings hostSettings, ReceiveSettings[] receivers, string[] sendingAddresses, CancellationToken cancellationToken = default)
         {
-            var baseTransportInfrastructure = baseTransport.Initialize(settings, connectionString);
-            return new ServerlessTransportInfrastructure<TBaseTransport>(baseTransportInfrastructure, settings);
+            //   hostSettings.
+            var baseTransportInfrastructure = await BaseTransport.Initialize(hostSettings, receivers, sendingAddresses, cancellationToken).ConfigureAwait(false);
+            var serverlessTransportInfrastructure = new ServerlessTransportInfrastructure<TBaseTransport>(baseTransportInfrastructure);
+            PipelineInvoker = (PipelineInvoker)serverlessTransportInfrastructure.Receivers[MainReceiverId];
```

Comment:
this will cause issues if we're trying to enable some `SendOnly` like scenarios where there is no main receiver

Reply:
Is there a test case I can set up?

Reply:
I've tried to create some here https://github.com/Particular/NServiceBus.AzureFunctions.InProcess.ServiceBus/pull/618 but I think we can scope this out of this PR for now. Would `SendOnly` work on the previous version or would that throw an exception too?

Reply:
I spiked a bit on this and got something functional.  I put up a PR once this one is merged.